stages/org.osbuild.dracut.conf

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python3
+"""
+Create dracut configuration files
+
+Creates dracut configuration files under `/usr/lib/dracut/dracut.conf.d/`.
+Only a subset of configuration options is supported, with the intention
+to provide functional parity with `org.osbuild.dracut` stage.
+
+Each stage option represents a "*.conf" file, which will be created. Its value
+is a dictionary with keys representing the allowed `dracut.conf` options
+representing the option value. The value type is specific to each configuration
+option and may be string, list of strings or boolean value.
+
+Constrains:
+  - At least one configuration file must be defined in the stage options.
+  - At least one configuration option must be specified for each configuration
+    file.
+
+Supported configuration options:
+  - compress
+  - dracutmodules
+  - add_dracutmodules
+  - omit_dracutmodules
+  - drivers
+  - add_drivers
+  - force_drivers
+  - filesystems
+  - install_items
+  - early_microcode
+  - reproducible
+"""
+
+import sys
+
+import osbuild.api
+
+
+SCHEMA = r"""
+"additionalProperties": false,
+"minProperties": 1,
+"patternProperties": {
+  "^[\\w.-]{1,250}\\.conf$": {
```

Comment:
I wonder, as with the `chmod` stage, if we should not move this into a one sub level, to allow us the flexibility to have other global options that we might want in the future (like the config dir path). Alternatively we could have:

```
{
  "filename": "",
  "config": {
    "compress": "..."
  }
}
```

And if you want more than one file, you add another `dracut.conf` stage. I think I personally prefer that since it is a tiny bit simpler, but OTOH we have done it differently now in some other stages.

Reply:
Right. I think this is a matter of preference, but I'm all for consistency if we decide which approach should be preferred. The reason why I'm choosing this approach is that it is more convenient from my point of view while iterating over all the configuration files and processing them. However that is really very minor "advantage" which is also still subjective.

As for moving the schema by one sub level, I commented in another comment below...

Reply:
Yeah, I agree, especially on the consistency domain. I _feel_ (hehe) having one stage write one feel is somehow nicer, makes the schema and the stage itself minimally easier (one less indirection) and even on the composer side one less nested struct. Hence my personal preference (`configuration_files` also seems verbose and generic, i.e. vs. `users` in `org.osbuild.user` top level).

But for consistency, maybe we should have the `configuration_files` key for now, and then change it in all the new stages in one PR, _if_ we decide for that.